c2corg_api/views/user_mailinglists.py

```diff
@@ -66,21 +70,23 @@ def post(self):
         subscribed_lists = DBSession.query(Mailinglist). \
             filter(Mailinglist.user_id == user_id).all()
         subscribed_lists = {l.listname: l for l in subscribed_lists}
-        subscribed_listnames = list(subscribed_lists.keys())
+        subscribed_listnames = set(subscribed_lists.keys())
 
         lists_to_add = []
         removed_lists = False
-        data = self.request.json_body
-        for listname in data:
-            if data[listname] and listname not in subscribed_listnames:
+        data = self.request.validated['mailinglists']
+        for listname in valid_mailinglists:
```

Comment:
I don't think this is correct. Here we should loop among the mailing lists provided by the request (ie. the actual changes to do), not among all the available mailing lists.

Reminder: the request provides only the ML whose status has changed. Eg. ``"avalanche": True, "lawinen": False``.